### PR TITLE
Issue 622: Only fully visible slides are interactable.

### LINKED ISSUE #622: Using tab to scroll pushes the slider to the left
# Bugs and Questions

### Prerequisites

**Feel free to delete this section if you have checked off all of the following.**

- [x] I've searched open [issues](https://github.com/FormidableLabs/nuka-carousel/issues) to make sure I'm not opening a duplicate issue
- [x] I have read through the docs before asking a question
- [x] I am using the latest version of nuka-carousel

### Describe Your Environment

- What version of nuka-carousel are you using?  4.5.13
- What version of React are you using? 16.9.0
- What browser are you using? Chrome
- What machine are you on? Macbook pro 15

### Describe the Problem

I cannot share my code but I dont think you will need it to understand the problem.

With a carousel showing multiple items, i use tab to check the accessibility in my page. It tabs on all slides and when it passes from the last visible slide to the ones that are hidden in the overflow, the .slider div is pushed on the left to show the hidden slides thus pushing the centerRightControl in the middle of the page where the last shown slide should be.

**Expected behavior:** Either when i tab the whole carousel is selected and the next tab skips to the DOM element after the carousel or I can tab only on the visible slides

**Actual behavior:** what i said earlier

### Additional Information

Expected:

carrousel_working_1

Reality: 

carrousel_notworking_2

### PR DESCRIPTION
**Pull request by mariano-formidable**
**Warning:** This PR adds the package `wicg-inert` which currently has a bug in it that causes issues with server side rendering.  There is  PR out to fix it, but it's not yet been merged (https://github.com/WICG/inert/pull/144).  Once _that_ fix is merged, this PR should update `wicg-inert` to the latest package and then this can be merged!

### Description

This is a second attempt for this PR to fix issue #622.  With the previous version, a user found an issue (#664) that fully visible slides were being marked as `inert` and thus weren't inter-actable.  I believe I fixed the logic in `isFullyVisible` so slides won't be incorrectly tagged as `inert`.

Fixes #622 #664 

#### Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

All existing tests pass and a new e2e test was written that failed at first, but now passes with the updated code.

### Screenshots
The below screenshots demonstrate that the anchor inside each slide is clickable when the slide is fully visible (the carousel title text toggles an underline as the anchors are clicked).

Slides remain active with cellspacing and multiple slides to show, left-aligned:
left_active_anchor

Right-aligned:
right_active_anchor

Center-aligned:
center_active_anchor

And works with wrap around enabled too:
wrapAround_active_anchor

### CODE REVIEW

File: src/utilities/utilities.js [hunk 1]
```diff
@@ -169,3 +170,49 @@ export const handleSelfFocus = e => {
...
+    slideWidth,
+    frameWidth,
+    wrapAround
+  } = config;
+
+  // Slide width can't be 0
+  const fullSlideWidth = slideWidth || 1;
+  // Calculate offset without cellSpacing
+  const offsetWidth =
+    getAlignmentOffset(currentSlide, config) + cellSpacing * currentSlide;
```
Comment (mariano-formidable) [inline src/utilities/utilities.js]: Adding back the cellSpacing offset fixed the bug where visible slides were being tagged as `inert` when cellSpacing was greater than 0.

File: test/e2e/carousel.test.js [hunk 2]
```diff
@@ -384,6 +384,47 @@ describe('Nuka Carousel', () => {
...
+      await page.mouse.click(pointX, pointY);
+      textDecoration = await page.evaluate(getTextDecoration);
+      expect(textDecoration).toMatch('none');
+
+      await expect(page).toClick('button', { text: 'Next' });
+      await page.waitFor(600); // need to let slide transition complete
+
+      await expect(page).toMatch('Nuka Carousel: Slide 2');
+
+      // Make should work on this page too
```
Comment (sarmeyer) [inline test/e2e/carousel.test.js]: missing a word in this comment I think

Comment (mariano-formidable) [reply test/e2e/carousel.test.js]: Whoops, good catch, I'll fix 😎

Comment (mariano-formidable) [reply test/e2e/carousel.test.js]: This should be fixed!

**Comment by YLemelin09** [comment]
I commented on the wicg-inert PR. There is no activity on it for 2 months. It seems it is just waiting for the approval of the 1 reviewer...

**Comment by YLemelin09** [comment]
Got things moving a bit for https://github.com/WICG/inert/pull/144

**Comment by YLemelin09** [comment]
wicg-inert just got updated to v3.0.3 with the fix

File: package.json [hunk 3]
```diff
@@ -11,7 +11,8 @@
     "d3-ease": "^1.0.3",
     "exenv": "^1.2.0",
     "prop-types": "^15.6.0",
-    "react-move": "^6.1.0"
+    "react-move": "^6.1.0",
+    "wicg-inert": "^3.0.3"
```
Comment (doxiaodong) [inline package.json:15]: `wicg-inert` v3 has es6 code  in src/inert.js. which used by module in package.json
see https://github.com/WICG/inert/blob/master/package.json#L6

Comment (rmillener) [reply package.json:15]: Just to add, this breaks IE11 even with Polyfills.

**Comment by YLemelin09** [comment]
Has this been tested using only tab to navigate in the carousel elements? I have just updated to 4.7.0 and Still can navigate past the slider div while "tabbing" thru the children elements of the carousel. Which results in the slider div and the controls to be pushed in the left when we get to the first child element past the slider div. 

Should i just try to have a way in my implementation to not assign a tab index on the elements outside of the slider div or you think it is a concern in the carousel itself?

**Pull request merged**